apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -409,6 +409,13 @@ type DatacenterOptions struct {
 	// ManagementApiAuth defines the authentication settings for the management API in the Cassandra pods.
 	// +optional
 	ManagementApiAuth *cassdcapi.ManagementApiAuthConfig `json:"managementApiAuth,omitempty"`
+
+	// The image to use in each Cassandra pod for the (short-lived) init container that merges global and perNodeConfig
+	// configuration. This is only useful when PerNodeConfigMapRef is set.
+	// The default is "mikefarah/yq:4".
+	// +optional
+	// +kubebuilder:default={repository:"mikefarah",name:"yq",tag:"4"}
+	PerNodeInitContainerImage *images.Image `json:"perNodeInitContainerImage,omitempty"`
```

Comment:
Hi @studypyth thanks for contributing to the K8ssandra project!

Two remarks on that line:

1. I suppose you declared this field here because, contrary to `PerNodeConfigMapRef`, you would like it to be declarable at both cluster and DC level? If that's why, I think that's fair, and a good idea.
2. However you used `images.Image` here, probably taking `JmxInitContainerImage` as a model. But we want to get rid of `images.Image` because it's too clunky, see https://github.com/k8ssandra/k8ssandra-operator/issues/532 for details. Given that, would you mind replacing `images.Image` with just a string?

Reply:
Also, I'm nitpicking but I'd prefer that you name your field `perNodeConfigInitContainerImage` – because otherwise the name is ambiguous and would mean "the image to use for per-node init-containers".

Reply:
Hi, @adutra, thank you for review.
1. Yes, exactly
2. I think that image.Image is the target type  according to medusa, stargate, jmxInitContainer. I have changed type to string, pls check.

Reply:
Thanks for the change. We need to unify the way we handle images in K8ssandra. The team will discuss this topic shortly and hopefully we'll arrive to a good, common way of doing that. In the meantime, string is perfect 👍